### ISSUE TITLE
VQEProblem on 9 qubits getting killed without error message

### ISSUE THREAD

**Issue by BenePoggel** [eclipse-qrisp/Qrisp#667]
### Priority

Medium

### Qrisp Version

0.8.2

### Python Version

3.11.13

### Operating System

Linux

### Short Description

My VQEProblem with a 9-qubit diagonal Hamiltonian cannot run, but gets killed without further error message. Edit: Jax/Pennylane clash could be resolved (Pennylane update missing), so that's not the cause. VQEProblem is still killed though.

### Steps to Reproduce

1. Run the below example (VQEProblem on 9 qubits, diagonal Hamiltonian).

```python
import numpy as np
from qrisp import QuantumVariable
from qrisp.vqe import VQEProblem
from qrisp.operators.qubit import Z
import sys

# Hamiltonian
H = (
    -160.325456118848 * Z(0)
    + 50.0 * Z(0) * Z(1)
    + 50.0 * Z(0) * Z(2)
    + 50.0 * Z(0) * Z(3)
    + 21.2266047150761 * Z(0) * Z(4)
    + 50.0 * Z(0) * Z(6)
    + 14.2222483680606 * Z(0) * Z(7)
    - 170.897706166273 * Z(1)
    + 50.0 * Z(1) * Z(2)
    + 21.2266047150761 * Z(1) * Z(3)
    + 50.0 * Z(1) * Z(4)
    + 21.2266047150761 * Z(1) * Z(5)
    + 14.2222483680606 * Z(1) * Z(6)
    + 50.0 * Z(1) * Z(7)
    + 14.2222483680606 * Z(1) * Z(8)
    - 160.325456118848 * Z(2)
    + 21.2266047150761 * Z(2) * Z(4)
    + 50.0 * Z(2) * Z(5)
    + 14.2222483680606 * Z(2) * Z(7)
    + 50.0 * Z(2) * Z(8)
    - 166.293549415572 * Z(3)
    + 50.0 * Z(3) * Z(4)
    + 50.0 * Z(3) * Z(5)
    + 50.0 * Z(3) * Z(6)
    + 16.0642145527729 * Z(3) * Z(7)
    - 174.581638535698 * Z(4)
    + 50.0 * Z(4) * Z(5)
    + 16.0642145527729 * Z(4) * Z(6)
    + 50.0 * Z(4) * Z(7)
    + 16.0642145527729 * Z(4) * Z(8)
    - 166.293549415572 * Z(5)
    + 16.0642145527729 * Z(5) * Z(7)
    + 50.0 * Z(5) * Z(8)
    - 135.298326507515 * Z(6)
    + 50.0 * Z(6) * Z(7)
    + 50.0 * Z(6) * Z(8)
    - 160.572925841667 * Z(7)
    + 50.0 * Z(7) * Z(8)
    - 135.298326507515 * Z(8)
)


def hea_ry_ansatz(qv, params):
    from qrisp import ry

    for i in range(qv.size):
        ry(params[i], qv[i])

qarg = QuantumVariable(9)
problem = VQEProblem(hamiltonian=H, ansatz_function=hea_ry_ansatz, num_params=9)
energy = problem.run(qarg, depth=1, max_iter=3)
print(f"SUCCESS. Energy: {energy}", flush=True)
```

### Expected Behavior

Either:
- VQEProblem completing the run, outputting some energy
Or:
- error message detailing the issue (memory overflow?)

### Actual Behavior

Process gets killed a few seconds after starting VQEProblem.run(). A RuntimeWarning detailing the dependency clash gets printed. QAOAProblem on the same problem runs without problems, so I suspect the error is linked to the expectation value estimation of the VQE operator.

Output:
WARNING:2026-06-19 11:45:13,384:jax._src.xla_bridge:864: An NVIDIA GPU may be present on this machine, but a CUDA-enabled jaxlib is not installed. Falling back to cpu.
/users/benedikt.poggel/miniforge3/envs/decisiontree-dev/lib/python3.11/site-packages/pennylane/__init__.py:196: RuntimeWarning: PennyLane is not yet compatible with JAX versions > 0.6.2. You have version 0.7.1 installed. Please downgrade JAX to 0.6.2 to avoid runtime errors using python -m pip install jax~=0.6.0 jaxlib~=0.6.0
  warnings.warn(
Killed

### Means of Verification

| Test-ID | Type     | Description | Expected Result |
|---------|----------|-------------|-----------------|
| T-001   | Positive |             |                 |
| T-002   | Negative |             |                 |
| T-003   | Boundary |             |                 |
| T-004   | Boundary |             |                 |

### Definition of Done

- [ ] Root cause identified
- [ ] Fix implemented and peer-reviewed
- [ ] Regression test added
- [ ] All tests pass (CI green)
- [ ] Changelog entry added

**Comment by renezander90** [comment]
Thanks for reporting this issue. The sum of the absolute values of the coefficients is quite large. This translates into a very large number of shots for default precision 0.01. I just checked that it uses 2_768_980_899 shots. We will further investigate. 
In the meantime normalizing the Hamiltonian will help:

```
norm = np.sum(np.abs(H.coeffs()))
H = 1.0/norm * H
```

**Comment by positr0nium** [comment]
Would it make sense to make the default precision some percentage of the norm?
